Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/misc.xml	(revision 673f546d77a1823724d330bfcdda0d468f24dfd1)
+++ .idea/misc.xml	(revision )
@@ -1,0 +1,0 @@
Index: app/src/main/res/layout/speechtotext.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/speechtotext.xml	(revision 673f546d77a1823724d330bfcdda0d468f24dfd1)
+++ app/src/main/res/layout/speechtotext.xml	(revision )
@@ -89,13 +89,14 @@
             tools:layout_constraintLeft_creator="1"
             android:layout_marginBottom="107dp"
             app:layout_constraintLeft_toLeftOf="parent"
-            android:layout_marginLeft="8dp"
+            android:layout_marginLeft="38dp"
             app:layout_constraintTop_toTopOf="parent"
             android:layout_marginTop="8dp"
             android:layout_marginRight="8dp"
             app:layout_constraintRight_toRightOf="parent"
-            app:layout_constraintHorizontal_bias="0.386"
-            app:layout_constraintVertical_bias="0.79" />
+            app:layout_constraintHorizontal_bias="0.379"
+            app:layout_constraintVertical_bias="0.736"
+            android:layout_marginEnd="8dp" />
 
         <ImageButton
             android:id="@+id/imageButton"
@@ -116,8 +117,8 @@
             app:layout_constraintHorizontal_bias="0.335"
             app:layout_constraintBottom_toBottomOf="parent"
             android:layout_marginBottom="8dp"
-            android:layout_marginLeft="8dp"
-            android:layout_marginRight="8dp"
+            android:layout_marginLeft="73dp"
+            android:layout_marginRight="75dp"
             app:layout_constraintVertical_bias="0.323" />
 
         <!--<TextView-->
@@ -147,13 +148,33 @@
 
             app:layout_constraintBottom_toBottomOf="parent"
             android:layout_marginBottom="8dp"
-            android:layout_marginRight="8dp"
-            app:layout_constraintRight_toRightOf="parent"
             android:layout_marginLeft="8dp"
             app:layout_constraintLeft_toLeftOf="parent"
             app:layout_constraintTop_toTopOf="parent"
             android:layout_marginTop="8dp"
-            app:layout_constraintVertical_bias="0.76" />
+            app:layout_constraintVertical_bias="0.757"
+            android:layout_marginStart="8dp" />
+
+        <Button
+            android:id="@+id/button3"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:background="?android:attr/textSelectHandle"
+            android:text="Entertainmernt"
+            android:visibility="gone"
+            tools:layout_editor_absoluteX="152dp"
+            tools:layout_editor_absoluteY="509dp" />
+
+        <Button
+            android:id="@+id/button7"
+            android:layout_width="88dp"
+            android:layout_height="48dp"
+            android:background="?android:attr/textSelectHandle"
+            android:onClick="foodBut"
+            android:text="Food"
+            android:visibility="invisible"
+            tools:layout_editor_absoluteX="123dp"
+            tools:layout_editor_absoluteY="450dp" />
 </android.support.constraint.ConstraintLayout>
 
 
Index: app/src/main/res/menu/billing_menu.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/menu/billing_menu.xml	(revision 673f546d77a1823724d330bfcdda0d468f24dfd1)
+++ app/src/main/res/menu/billing_menu.xml	(revision )
@@ -8,6 +8,10 @@
     <item
         android:id="@+id/action_name"
         android:title="@string/name_billing"
+
+
+        android:textAlignment="center"
+
         app:showAsAction="always" />
 
     <item
\ No newline at end of file
Index: app/src/main/java/com/wanmoon/finwal/activity/SpeechToText.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wanmoon/finwal/activity/SpeechToText.java	(revision 673f546d77a1823724d330bfcdda0d468f24dfd1)
+++ app/src/main/java/com/wanmoon/finwal/activity/SpeechToText.java	(revision )
@@ -11,14 +11,14 @@
 import android.widget.ImageButton;
 import android.widget.TextView;
 import android.widget.Toast;
-
+import java.util.Arrays;
 import com.google.firebase.database.DataSnapshot;
 import com.google.firebase.database.DatabaseError;
 import com.google.firebase.database.DatabaseReference;
 import com.google.firebase.database.FirebaseDatabase;
 import com.google.firebase.database.ValueEventListener;
 import com.wanmoon.finwal.R;
-
+import java.util.List;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
@@ -37,8 +37,13 @@
     private static String val = "";
     private TextView textViewFinish;
     private TextView textViewCancel;
+    private TextView textViewFood;
+    private TextView textViewEnt;
+    private TextView textViewEdu;
+    private TextView textViewShop;
 
 
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
@@ -51,6 +56,7 @@
         textViewCancel = (TextView)findViewById(R.id.textViewCancel);
         textViewFinish.setOnClickListener(this);
         textViewCancel.setOnClickListener(this);
+        textViewFood = (TextView)findViewById(R.id.button7);
 
 
     }
@@ -129,21 +135,21 @@
 
                         DatabaseReference databaseReference = database.getReference();
                         databaseReference.child("Category").child("" + test[j]).child("caType").addListenerForSingleValueEvent(new ValueEventListener() {
-                                                                                                                                   @Override
-                                                                                                                                   public void onDataChange(DataSnapshot dataSnapshot) {
-                                                                                                                                       String value = dataSnapshot.getValue(String.class);
-                                                                                                                                       textViewStatus.setText("Cetegory is " + value);
-                                                                                                                                       Log.d("", "value is" + value);
-                                                                                                                                   }
+                            @Override
+                            public void onDataChange(DataSnapshot dataSnapshot) {
+                                String value = dataSnapshot.getValue(String.class);
+                                textViewStatus.setText("Cetegory is " + value);
+                                Log.d("", "value is" + value);
+                            }
-
-                                                                                                                                   @Override
-                                                                                                                                   public void onCancelled(DatabaseError databaseError) {
-                                                                                                                                       Log.w("", "Failed to read value.");
-                                                                                                                                   }
-                                                                                                                               }
+                            @Override
+                            public void onCancelled(DatabaseError databaseError) {
+                                Log.w("", "Failed to read value.");
+                            }
+                        }
-
                         );
                     }
+                    textViewFood.setVisibility(View.VISIBLE);
+
                 }
 
             }
@@ -156,19 +162,14 @@
     @Override
     public void onClick(View v) {
         if(v == textViewFinish){
-
-             //will open login activity here
+            //will open login activity here
             Intent i=new Intent(getApplicationContext(), MainActivity.class);
             startActivity(i);
-
-
         }
         if(v == textViewCancel){
             // will open login activity here
             Intent i=new Intent(getApplicationContext(), MainActivity.class);
             startActivity(i);
-
-
         }
     }
     public List keepPrice(){
\ No newline at end of file
