Index: app/src/main/res/menu/billing_menu.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/menu/billing_menu.xml	(revision fb84a617e1e1b416c24839a4a45e6f86c0856707)
+++ app/src/main/res/menu/billing_menu.xml	(revision )
@@ -7,13 +7,12 @@
 
     <item
         android:id="@+id/action_name"
-        android:title="@string/name_billing"
+        android:textAlignment="center"
 
 
-        android:textAlignment="center"
+        android:title="@string/name_billing"
 
         app:showAsAction="always" />
-
     <item
         android:id="@+id/action_add"
         android:icon="@mipmap/ic_add_white_24dp"
\ No newline at end of file
Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/modules.xml	(revision fb84a617e1e1b416c24839a4a45e6f86c0856707)
+++ .idea/modules.xml	(revision )
@@ -3,7 +3,6 @@
   <component name="ProjectModuleManager">
     <modules>
       <module fileurl="file://$PROJECT_DIR$/FinWal.iml" filepath="$PROJECT_DIR$/FinWal.iml" />
-      <module fileurl="file://$PROJECT_DIR$/FinWal.iml" filepath="$PROJECT_DIR$/FinWal.iml" />
       <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
     </modules>
   </component>
\ No newline at end of file
Index: app/src/main/java/com/wanmoon/finwal/activity/SpeechToText.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wanmoon/finwal/activity/SpeechToText.java	(revision fb84a617e1e1b416c24839a4a45e6f86c0856707)
+++ app/src/main/java/com/wanmoon/finwal/activity/SpeechToText.java	(revision )
@@ -11,14 +11,14 @@
 import android.widget.ImageButton;
 import android.widget.TextView;
 import android.widget.Toast;
-
+import java.util.Arrays;
 import com.google.firebase.database.DataSnapshot;
 import com.google.firebase.database.DatabaseError;
 import com.google.firebase.database.DatabaseReference;
 import com.google.firebase.database.FirebaseDatabase;
 import com.google.firebase.database.ValueEventListener;
 import com.wanmoon.finwal.R;
-
+import java.util.List;
 import java.util.ArrayList;
 import java.util.Locale;
 
@@ -26,12 +26,13 @@
  * Created by pimpischaya on 5/27/2017 AD.
  */
 
+
 public class SpeechToText extends AppCompatActivity implements View.OnClickListener{
     private TextView textViewStatus;
     private EditText btnInput;
     private TextView resultTEXT;
     private ImageButton imageButton;
-
+    private static String val = "";
     private TextView textViewFinish;
     private TextView textViewCancel;
 
@@ -110,6 +111,7 @@
     public void onActivityResult(int request_code , int result_code , Intent i){
         FirebaseDatabase database = FirebaseDatabase.getInstance();
         textViewStatus = (TextView)findViewById(R.id.textViewStatus);
+
         super.onActivityResult(request_code,result_code,i);
 
         switch(request_code){
@@ -117,27 +119,36 @@
             {
                 ArrayList<String> result = i.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS);
                 resultTEXT.setText(result.get(0));
-                String val = result.get(0).toString();
+
+                val = result.get(0).toString();
+                String[] test = {"ค่าไฟ","ต้มยำ"};
+                for(int j=0;j<test.length;j++) {
+                    if (val.matches(".*" + test[j] + ".*") == true) {
+
-                DatabaseReference databaseReference = database.getReference();
+                        DatabaseReference databaseReference = database.getReference();
-                databaseReference.child("Category").child(""+val).child("caType").addListenerForSingleValueEvent(new ValueEventListener() {
+                        databaseReference.child("Category").child("" + test[j]).child("caType").addListenerForSingleValueEvent(new ValueEventListener() {
-                                                                          @Override
-                                                                          public void onDataChange(DataSnapshot dataSnapshot) {
-                                                                              String value = dataSnapshot.getValue(String.class);
-                                                                              textViewStatus.setText("Cetegory is " + value);
-                                                                              Log.d("", "value is" + value);
-                                                                          }
+                                                                                                                                   @Override
+                                                                                                                                   public void onDataChange(DataSnapshot dataSnapshot) {
+                                                                                                                                       String value = dataSnapshot.getValue(String.class);
+                                                                                                                                       textViewStatus.setText("Cetegory is " + value);
+                                                                                                                                       Log.d("", "value is" + value);
+                                                                                                                                   }
 
-                                                                          @Override
-                                                                          public void onCancelled(DatabaseError databaseError) {
-                                                                              Log.w("", "Failed to read value.");
-                                                                          }
-                                                                      }
+                                                                                                                                   @Override
+                                                                                                                                   public void onCancelled(DatabaseError databaseError) {
+                                                                                                                                       Log.w("", "Failed to read value.");
+                                                                                                                                   }
+                                                                                                                               }
+
-                );
+                        );
+                    }
+                }
 
             }
 
                 break;
         }
+
     }
 
     @Override
@@ -156,5 +167,9 @@
 
         }
     }
-
+    public List keepPrice(){
+        val = val.replaceAll("[^0-9]+", " ");
+        List price = Arrays.asList(val.trim().split(" "));
+        return price;
+    }
-}
+}
\ No newline at end of file
