Index: app/src/main/res/layout/activity_main.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_main.xml	(revision c11da6c7a28619d6f859061b113e6d3f15b0a91c)
+++ app/src/main/res/layout/activity_main.xml	(revision )
@@ -9,6 +9,7 @@
     android:fitsSystemWindows="true"
     tools:openDrawer="start">
 
+
     <include
         layout="@layout/app_bar_main"
         android:layout_width="match_parent"
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/build.gradle	(revision c11da6c7a28619d6f859061b113e6d3f15b0a91c)
+++ app/build.gradle	(revision )
@@ -3,45 +3,57 @@
 apply plugin: 'com.google.gms.google-services'
 
 android {
+
     compileSdkVersion 25
     buildToolsVersion "25.0.2"
     defaultConfig {
         applicationId "com.wanmoon.finwal"
+       // manifestPlaceholders = [onesignal_app_id: "68819f6e-cb54-45e4-850d-5c55d334ebcd",
+                                // Project number pulled from dashboard, local value is ignored.
+       //                         onesignal_google_project_number: "REMOTE"]
         minSdkVersion 22
         targetSdkVersion 25
         versionCode 1
         versionName "1.0"
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
-    }
+
-    buildTypes {
-        release {
-            minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
-        }
-    }
+        buildTypes {
+            release {
+                minifyEnabled false
+                proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+            }
+        }
+
+
+
-    productFlavors {
-    }
-}
+        productFlavors {
+        }
+    }
 
-dependencies {
+    dependencies {
+       // compile 'com.onesignal:OneSignal:[3.6.1,3.99.99)'
+        compile 'com.akexorcist:RoundCornerProgressBar:2.0.3'
+
-    compile fileTree(include: ['*.jar'], dir: 'libs')
-    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
-        exclude group: 'com.android.support', module: 'support-annotations'
-    })
+        compile fileTree(include: ['*.jar'], dir: 'libs')
+        androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
+            exclude group: 'com.android.support', module: 'support-annotations'
+        })
 
+        //compile 'com.google.android.gms:play-services-gcm:11.0.4'
 
-
-    compile 'com.akexorcist:RoundCornerProgressBar:2.0.3'
-    compile 'com.squareup.okhttp3:okhttp:3.8.1'
-    compile 'com.google.code.gson:gson:2.6.1'
-    compile 'com.android.support:appcompat-v7:25.3.1'
-    compile 'com.android.support:design:25.1.0'
-    compile 'com.google.firebase:firebase-auth:10.0.1'
-    compile 'com.google.firebase:firebase-database:10.0.1'
-    compile 'com.android.support.constraint:constraint-layout:1.0.2'
-    compile 'com.android.support:support-v4:25.3.1'
-    compile 'org.achartengine:achartengine:1.2.0'
-    compile 'com.github.PhilJay:MPAndroidChart:v3.0.0-beta1'
-    compile 'com.jjoe64:graphview:4.2.1'
+        compile 'com.akexorcist:RoundCornerProgressBar:2.0.3'
+        compile 'com.squareup.okhttp3:okhttp:3.8.1'
+        compile 'com.google.code.gson:gson:2.6.1'
+        compile 'com.android.support:appcompat-v7:25.3.1'
+        compile 'com.android.support:design:25.1.0'
+        compile 'com.google.firebase:firebase-auth:10.0.1'
+        compile 'com.google.firebase:firebase-database:10.0.1'
+        compile 'com.android.support.constraint:constraint-layout:1.0.2'
+        compile 'com.android.support:support-v4:25.3.1'
+        compile 'org.achartengine:achartengine:1.2.0'
+        compile 'com.github.PhilJay:MPAndroidChart:v3.0.0-beta1'
+        compile 'com.jjoe64:graphview:4.2.1'
+
-    testCompile 'junit:junit:4.12'
+        testCompile 'junit:junit:4.12'
+    }
-}
+}
\ No newline at end of file
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- build.gradle	(revision c11da6c7a28619d6f859061b113e6d3f15b0a91c)
+++ build.gradle	(revision )
@@ -9,6 +9,7 @@
         classpath 'com.google.gms:google-services:3.0.0'
 
 
+
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
     }
@@ -19,6 +20,8 @@
         jcenter()
         maven { url "https://repository-achartengine.forge.cloudbees.com/snapshot/" }
         maven { url "https://jitpack.io" }
+        //maven { url "https://maven.google.com" }
+
 
     }
 }
Index: app/src/main/java/com/wanmoon/finwal/activity/Home.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wanmoon/finwal/activity/Home.java	(revision c11da6c7a28619d6f859061b113e6d3f15b0a91c)
+++ app/src/main/java/com/wanmoon/finwal/activity/Home.java	(revision )
@@ -1,6 +1,7 @@
 package com.wanmoon.finwal.activity;
 
 import android.content.Context;
+
 import android.content.Intent;
 import android.graphics.Color;
 import android.net.Uri;
@@ -138,6 +139,7 @@
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
 
         if (getArguments() != null) {
             mParam1 = getArguments().getString(ARG_PARAM1);
\ No newline at end of file
Index: app/src/main/java/com/wanmoon/finwal/activity/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wanmoon/finwal/activity/MainActivity.java	(revision c11da6c7a28619d6f859061b113e6d3f15b0a91c)
+++ app/src/main/java/com/wanmoon/finwal/activity/MainActivity.java	(revision )
@@ -1,6 +1,7 @@
 package com.wanmoon.finwal.activity;
-
 import android.content.Context;
+//import com.onesignal.*;
+//import com.onesignal.OneSignal.NotificationOpenedHandler;
 import android.content.Intent;
 import android.net.Uri;
 import android.os.Bundle;
@@ -26,6 +27,8 @@
 import com.google.firebase.auth.FirebaseUser;
 import com.wanmoon.finwal.R;
 
+import org.json.JSONObject;
+
 import java.io.IOException;
 
 import okhttp3.Call;
@@ -79,6 +82,14 @@
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        //Log.e("tet", Locale.getDefault().getLanguage());
+
+        //OneSignal.setInFocusDisplaying(OneSignal.OSInFocusDisplayOption.InAppAlert);
+        //OneSignal.sendTag("test3", "test7");
+        //OneSignal.setSubscription(false);
+
+        //OneSignal.syncHashedEmail("test@onesignal.com");
+
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
 
@@ -566,4 +577,68 @@
 
 
     }
+//    private class ExampleNotificationOpenedHandler implements NotificationOpenedHandler {
+//        /**
+//         * Callback to implement in your app to handle when a notification is opened from the Android status bar or
+//         * a new one comes in while the app is running.
+//         * This method is located in this activity as an example, you may have any class you wish implement NotificationOpenedHandler and define this method.
+//         *
+//         * @param openedResult The message string the user seen/should see in the Android status bar.
+//         */
+//        @Override
+//        public void notificationOpened(OSNotificationOpenResult openedResult) {
+//            Log.e("OneSignalExample", "body: " + openedResult.notification.payload.body);
+//            Log.e("OneSignalExample", "additional data: " + openedResult.notification.payload.additionalData);
+//            //Log.e("OneSignalExample", "additionalData: " + additionalData.toString());
+//        }
+//    }
+//public void onSubscribeClicked(View v) {
+//    OneSignal.setSubscription(true);
+//    OneSignal.promptLocation();
+//
+//    String ns = Context.NOTIFICATION_SERVICE;
+//    NotificationManager nMgr = (NotificationManager) getApplicationContext().getSystemService(ns);
+//    nMgr.cancelAll();
+//}
+//
+//    public void onUnsubscribeClicked(View v) {
+//        OneSignal.setSubscription(false);
+//    }
+
+//   @Override
+//   protected void onPause() {
+//      super.onPause();
+//      if (mHelper != null)
+//         mHelper.dispose();
+//   }
+//
+//   @Override
+//   public void onDestroy() {
+//      super.onDestroy();
+//      if (mHelper != null) mHelper.dispose();
+//      mHelper = null;
+//   }
+
+//    @Override
+//    public boolean onCreateOptionsMenu(Menu menu) {
+//        // Inflate the menu; this adds items to the action bar if it is present.
+//        getMenuInflater().inflate(R.menu.menu_main, menu);
+//        return true;
+//    }
+//
+//    @Override
+//    public boolean onOptionsItemSelected(MenuItem item) {
+//        // Handle action bar item clicks here. The action bar will
+//        // automatically handle clicks on the Home/Up button, so long
+//        // as you specify a parent activity in AndroidManifest.xml.
+//        int id = item.getItemId();
+//
+//        //noinspection SimplifiableIfStatement
+//        if (id == R.id.action_settings)
+//            return true;
+//
+//        return super.onOptionsItemSelected(item);
+//    }
+
+
 }
